File: 3001_6000/3986.py
ans = 0
for _ in [0]*int(input()):
    s = input()
    l = []
    for i in s:
        if len(l)==0 or l[-1] != i:
            l.append(i)
        else:
            l.pop(len(l)-1)
    if len(l)==0 : ans+=1
print(ans)

File: 3001_6000/3986.cpp
#include<iostream>
#include<stack>
#include<string>
using namespace std;

int main()
{
    int N;
    cin>>N;
    string input;
    int output = 0;
    for(int n = 0; n<N; n++)
    {
        stack<char> s;
        cin>>input;
        for(auto &i:input)
        {
            if(s.empty() || s.top() != i)
                s.push(i);
            else
                s.pop();
        }
        if(s.empty()) output++;
    }
    cout<<output<<endl;
}